src/components/component/secrets/secret-list-item-title-azure-key-vault-item.tsx

```diff
@@ -12,6 +12,27 @@ import { RequestState } from '../../../state/state-utils/request-states';
 
 import '../style.css';
 
+function getSecretInReplicaWithUniqueBatchNames(secretInReplicaList) {
+  let batchMap = new Map<string, Map<string, boolean>>(); //batchName:version:boolean
+  // this filtering is to avoid showing all job pods for the same batch with the same secret version
+  return secretInReplicaList.filter((secretInReplica) => {
+    if (
+      secretInReplica.batchName === null ||
+      secretInReplica.batchName.length === 0
+    )
```

Comment:
null check does not capture undefined
```suggestion
    if (!(secretInReplica.batchName?.length > 0))
```